### ISSUE TITLE
[BUG] On-screen Layer info not working printing from onboard SD

### ISSUE THREAD

**Issue by blueeagle69** [bigtreetech/BIGTREETECH-TouchScreenFirmware#1135]
Hi.

When using the .26 firmware (latest build) and the build from the Biqu B1 repo the on-screen layer information is not working when printing from the onboard SD.

However if I print from any media connected to the TFT it works as expected.

My screen is a Biqu custom TFT35 e3 V3.

Thanks.

**Comment by oldman4U** [comment]
Please add the information of the Slicer and which firmware worked well (for some time at least...)

**Comment by Sebazzz** [comment]
I have the same issue. Cura 4.6.2 and Marlin 2.0.6 - although I have ported all changes from the BIQU fork from Marlin. I will reflash an older Marlin when I have the time and report back.

**Comment by blueeagle69** [comment]
> Please add the information of the Slicer and which firmware worked well (for some time at least...)

I'm using Prusaslicer, with;
;LAYER: [layer_num]
;Z: [layer_z]
in Before_Layer_Change.

I'm sure now the only reason it worked before was due to the fact I was printing from media in the TFT.
If you print from the onboard (SKR1.4) SD card it doesn't work.

**Comment by Sebazzz** [comment]
Is the number the same as the Z position in emulation mode? Because that does work - even when printing from the SD card in the printer.

Met vriendelijke groet,
Sebastiaan Dammann
________________________________
Van: blueeagle69 <notifications@github.com>
Verzonden: Wednesday, September 30, 2020 4:15:33 PM
Aan: bigtreetech/BIGTREETECH-TouchScreenFirmware <BIGTREETECH-TouchScreenFirmware@noreply.github.com>
CC: Sebastiaan Dammann <sebastiaandammann@outlook.com>; Manual <manual@noreply.github.com>
Onderwerp: Re: [bigtreetech/BIGTREETECH-TouchScreenFirmware] [BUG] On-screen Layer info not working printing from onboard SD (#1135)


Please add the information of the Slicer and which firmware worked well (for some time at least...)

I'm using Prusaslicer, with;
;LAYER: [layer_num]
;Z: [layer_z]
in Before_Layer_Change.

I'm sure now the only reason it worked before was due to the fact I was printing from media in the TFT.
If you print from the onboard (SKR1.4) SD card it doesn't work.

**Comment by blueeagle69** [comment]
At the moment I'm stuck on an older .26 build from the B1 repo due to the screen switching from Marlin to TFT when I start a print, so I can't confirm.  But I know on my current B1 build if I print from the tft card the layer info shows firstly the layer height (0.200mm) and increases every layer afterwards.  But from the onboard it stays at 0.00mm. If I could stop my screen switching from Marlin mode to tft and locking me there I would re update and check . I mostly use Marlin mode so I can use  my smart filament sensor.

**Comment by oldman4U** [comment]
@Sebazzz 
The problem is in the slicer and Marlin. Cura gcode should work straight forward. Somewhere I saw information how to change Marlin to make this work but can not find it anymore.

**Comment by blueeagle69** [comment]
Just to clarify.
I'm currently running firmware dated 220-8-18 due to the #1058 bug.

With this firmware I have sliced using Cura and Prusaslicer.
But when printing from the printer onboard SD the layer information never updates.
If this has been fixed in recent updates then apologies.

**Comment by oldman4U** [comment]
Marlin mode or Touchscreen? Can you attach a picture how this looks like on screen and also Menu-Settings-Info from TFT.

**Comment by blueeagle69** [comment]
In Touch mode.
Here are the screenshots.
![20201006_221019](https://user-images.githubusercontent.com/9963876/95261562-851b4800-0822-11eb-8d38-0f94db8548f9.jpg)
![20201006_221915](https://user-images.githubusercontent.com/9963876/95261589-92d0cd80-0822-11eb-99e7-e89c1c0d4360.jpg)

**Comment by Maker-M** [comment]
> In Touch mode.
> Works fine from TFT media but not onboard (mainboard).

Up, help me !

**Comment by oldman4U** [comment]
I have to correct myself from above (sorry!!!) and can confirm this behaviour.

In addition changing the temperature while preheating is ignored (M109 vs. M104).

Wouldn't it be good to make one (Known Issue) ticket for all the issues related to printing from OnBoard SD card?

**Comment by traffic-light** [comment]
This is a odd bug that's been there for a while. I'll never print from mainboard sd card.

One other thing: you're using the wrong firmware for your tft. It's telling you're using a tft35 e3. That's not the one you have.

**Comment by traffic-light** [comment]
> I have to correct myself from above (sorry!!!) and can confirm this behaviour.
> 
> In addition changing the temperature while preheating is ignored (M109 vs. M104).
> 
> Wouldn't it be good to make one (Known Issue) ticket for all the issues related to printing from OnBoard SD card?

That could be something for in the readme first. 
I'm almost done with my exams. Then I got a little more space to breath

**Comment by oldman4U** [comment]
Its already march!!??

👍🏻

**Comment by traffic-light** [comment]
Not yet. But in march I've even more time :-)

**Comment by oldman4U** [comment]
Sounds promising. 😁

**Comment by blueeagle69** [comment]
> This is a odd bug that's been there for a while. I'll never print from mainboard sd card.
> 
> One other thing: you're using the wrong firmware for your tft. It's telling you're using a tft35 e3. That's not the one you have.

Weird, that's the one Biqu told me to use with the Biqu B1. I have tried both though TFT35 V3 and the e3 variant.
I print from the onboard SD so I can slice straight to the card without  having to take it out. Especially as the B1 insists on using micro SD cards.

**Comment by blueeagle69** [comment]
I will tidy up my end and close my issues/bugs that you have kindly added to the pinned list 😀

**Comment by oldman4U** [comment]
👍🏻

blueeagle69 <notifications@github.com> schrieb am Do. 5. Nov. 2020 um 15:20:

**Comment by sadbass** [comment]
There is some news for this problem?

**Comment by oldman4U** [comment]
In contrast to what I wrote above, I believe this is a bit by design. Since I have the touchscreen TFT, I mainly print from the U disk of the TFT and so this all works perfectly. But before I bought the TFT, I used the standard Ender 3 screen and using this screen there was also no layer hight information coming from Marlin. So Marlin by default seems NOT to know anything about the layer and the layer hight it is printing at the moment. To make this work under Marlin, an extension has to be activated in Cura, where this information is written into the gcode and read out from there by Marlin.

So the question is, how the TFT can get this information while printing from the SD card of the mainboard, and therefore depending on the information which comes from Marlin.

My 2 cents

**Comment by oldman4U** [comment]
Please check PR #1272 to see if it solves the reported issue.

**Comment by blueeagle69** [comment]
Cheers @oldman4U I will compile and flash tonight or tomorrow and report, unless someone beats me to it!

**Comment by oldman4U** [comment]
👍🏻

blueeagle69 <notifications@github.com> schrieb am Do. 19. Nov. 2020 um
21:39:

**Comment by blueeagle69** [comment]
@oldman4U 
Layer info now works perfectly with #1272 when printing from onboard SD!
Also warmup temps are also working perfectly with the previous commit.
I have noticed though that if there is no mesh saved (ABL not active) the ABL icon does not show in the Movement screen.
If I go to Marlin mode and run ABL and save the EEPROM, when I go back to TFT mode the ABL icon appears.
I'm pretty-sure this was mentioned elsewhere.
Also, if you cancel an onboard SD card print you get the black Loading screen forever again.
I'm pretty-sure this was fixed, but it appears to have crept back in.
Not sure when though.
